### DISCUSSION TITLE
FCPX - Trim first XX frames of multiple clips in timeline

### DISCUSSION THREAD

**Discussion by cheapest-energy-windows** [CommandPost/CommandPost#3176] [Lua Snippet Questions] [answered]
In my workflow I always need to trim the first 90-100 frames of each clip on the timeline, usually more than 20. 

1) I can do this manually
2) I have a stream deck and with command post I have to buttons, one with forward playhead 90 frames and trim to playhead at release and one with go to next edit

I'm looking for a way where I could just select all clips, and trim all of them at the same time with e.g. 90 frames at the head. Pretty much something like how Control-D let's you do this at the end of each clip. 

Any suggestions if this would be possible?

Thanks

**Comment by randomeizer** [comment]
I think it would be doable with a snippet. @latenitefilms may be able to whip one up before I can.

**Comment by latenitefilms** [comment, marked as answer]
This is pretty rough, and might be unreliable, however, this will hopefully point you in the right direction:

```lua
local fcp               = require "cp.apple.finalcutpro"
local eventtap          = require "hs.eventtap"
local keyStroke         = eventtap.keyStroke

fcp:launch()

local contents = fcp.timeline.contents
local selectedClipsUI = contents:selectedClipsUI()
if #selectedClipsUI == 0 then
    cp.tools.playErrorSound()
    return
end

table.sort(selectedClipsUI, function(a, b) return a.AXPosition.x < b.AXPosition.x end)

for _, clip in pairs(selectedClipsUI) do
    contents:selectClip(clip)    
    keyStroke({"shift"}, "i")
    keyStroke({}, "]")
    keyStroke({}, "=")
    keyStroke({}, "3")
    keyStroke({}, "0")
    keyStroke({}, "0")
    keyStroke({}, "return")
    keyStroke({"option"}, "[")        
end
```

**Reply by cheapest-energy-windows** [reply]
Thank you so much, highly appreciated. Going to give this a try!

**Comment by terrylb** [comment]
I know this is very late and not a Lua solution, but how about: select all the clips, reverse them, change their duration by -90, then reverse them again.
